src/scheduler.h

```diff
@@ -187,6 +187,12 @@ class SchedulerService : public Scheduler::Service {
   Synchronized<std::vector<RefCount> > reference_counts_;
   // contained_objrefs_[objref] is a vector of all of the objrefs contained inside the object referred to by objref
   Synchronized<std::vector<std::vector<ObjRef> > > contained_objrefs_;
+  // All of the remote functions that have been exported to the workers. This
+  // data structure is protected by workers_.get().
+  std::vector<std::unique_ptr<Function> > exported_functions;
+  // All of the reusable variables that have been exported to the workers. This
+  // data structure is protected by workers_.get().
+  std::vector<std::unique_ptr<ReusableVar> > exported_reusable_variables;
```

Comment:
If this should really share a lock with `workers_`, then put all of them into their own struct and then protect *that* with `Synchronized`. You should not have dangling fields here.